src/System.Private.CoreLib/shared/System/Reflection/TypeDelegator.cs

```diff
@@ -21,11 +22,11 @@ public override bool IsAssignableFrom(TypeInfo typeInfo)
 
         protected Type typeImpl;
 
-        protected TypeDelegator() { }
+ //       protected TypeDelegator() { }
```

Comment:
Even `typeImpl` variable is initiated below construnctor and this constractor is protected having compiler error here. As there is lot of references/usage of `typeImpl` without null check all over this class I don't want to mark the variable nullable, would like to remove default contructor... there is no derived classes at least in coreclr.

Reply:
This would be a breaking change because we already exposed it to the world since .netstandard2.0 -- can we just initialize `typeImpl = null!` ? However since it is protected, we would be lying to whoever derives from this class. cc: @stephentoub making it nullable would be more honest?

Reply:
As @safern says, this is breaking change; please do not remove the parameterless protected ctor.

I'm ok with `protected Type typeImpl = null!` in this case.  Deriving from TypeDelegator is a super-advanced use case, and in situations where you do, you're responsible for supplying the `typeImpl`, either by setting it or by passing it to the ctor, so if you read it out and expect it to be non-null and dereference it, you probably shouldn't be using this type :)

Reply:
> so if you read it out and expect it to be non-null and dereference it, you probably shouldn't be using this type :)

🤣